### ISSUE TITLE
Could not connect to SignalR, UI won't update

### ISSUE THREAD

**Issue by Stealth103** [Radarr/Radarr#7597]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

I'm getting an error: Could not connect to SignalR, UI won't update.

### Expected Behavior

Not receiving the Could not connect to SignalR, UI won't update error after changing the authentication method.

### Steps To Reproduce

Switch authentication method from Forms to Basic under Settings -> General -> Security and then I received this error.  If it switch back from Basic to Forms the issue resolves itself.

### Environment

```markdown
- OS: Ubuntu 20.04.5
- Radarr: 4.1.0.6175
- Docker Install: Yes
- Using Reverse Proxy: No
- Browser: N/A
- Database: N/A
```

### What branch are you running?

Master

### Trace Logs?

https://pastebin.com/LVzju37q

[AB#3740](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/3740)

**Comment by ta264** [comment]
What browser?

**Comment by Stealth103** [comment]
![Radarr-Forms](https://user-images.githubusercontent.com/20715751/190473026-48ecc5af-7727-48fe-86ae-e7dbf16089fb.png)
![Radarr-Basic](https://user-images.githubusercontent.com/20715751/190473030-ab09f7f5-f805-4218-b9b4-1d05f054f011.png)
Here are some screenshots from before and after.

I'm using Brave Browser v1.43.93

**Comment by ta264** [comment]
Try Firefox/ chrome

**Comment by ta264** [comment]
And also try refreshing the page after changing the setting

**Comment by Stealth103** [comment]
I usually reboot after changing the setting, I've definitely tried refreshing.  I'm getting the same issue with Chrome Version 105.0.5195.127 (Official Build) (64-bit)

**Comment by ta264** [comment]
Are you using a reverse proxy? What URL do you use to access radarr?

**Comment by Stealth103** [comment]
No reverse proxy.  I use a local IP on my network: 192.168.xxx.xxx:7878

**Comment by Stealth103** [comment]
sorry should say http://192.168.xxx.xxx:7878

**Comment by Stealth103** [comment]
Also the same thing seems to be happening with Lidarr, it is not happening with my Sonarr instance.

**Comment by bakerboy448** [comment]
What browser extensions? Tried incognito Does the browser dev console (warning: not cleansed) or network calls give any indication what doesn't work?

**Comment by bakerboy448** [comment]
@Stealth103 any updates?

**Comment by Stealth103** [comment]
I'll take a look on my PC tomorrow.  It's happening on my phone as well running chrome or brave on android.  No extensions on my phone browser.

**Comment by bakerboy448** [comment]
pastebin expired btw

and yeah check the browser dev tools as well

**Comment by Stealth103** [comment]
I think I found where the error is happening, if you need a new trace file let me know, but I don't see anything that looks off in the trace file itself.  I removed the username and password information (and the access token where it did appear) from the output (replacing it with user:password).

https://logs.notifiarr.com/?d92433d35af70710#GkdxkFuEUBcumpMomXrnhXnfZsBbaS9CWDTtA8wEcUXn

**Comment by Stealth103** [comment]
From the output it seems that having my username/password credentials in the URL is causing the issue.

**Comment by bakerboy448** [comment]
```
Warning: Error from HTTP request. TypeError: Failed to execute 'fetch' on 'Window': Request cannot be constructed from a URL that includes credentials: http://user:password@192.168.10.12:7878/signalr/messages/negotiate?access_token=(removed)&negotiateVersion=1.
(anonymous) @ instrument.js:109
instrument.js:109
```

From some searching it seems fetch does not support credentials in the URL.

Passing your credentials in plaintext in the URL also pretty much negates any and all security and makes having the auth rather insecure given the credentials are not secured.

**Comment by unraidr** [comment]
SignalR works with basic auth embedded in the URL for Sonarr but not with Radarr, so where's the difference coming from?

> Passing your credentials in plaintext in the URL also pretty much negates any and all security and makes having the auth rather insecure given the credentials are not secured.

If using an SSL connection, those credentials are secured.

**Comment by bakerboy448** [comment]
Sonarr v3 is a completely different backend than Sonarr v4, Radarr, and every other \*Arr.    .NET framework 4.5 vs. .NET 6

**Comment by unraidr** [comment]
I agree the credentials aren't secure for a person that has access to the computer, but for the purposes of internet traffic, those credentials are secured if using an SSL cert as the credentials are parsed into the request headers and are encrypted. I just want to distinguish that from the general sentiment that it's "not secure".

> they're not secured - they're presumably stored plaintext in your browser history at minimum, but that's neither here nor there.

I just checked my Chrome history SQLite db and the credentials are not in the database, this is presumably due to the fact that Chrome would parse this into a request and only store the URL. Feel free to correct if you've seen something else.
